### PR TITLE
Integration tests

### PR DESCRIPTION
**Pull request by seriva**
Adds basic integration tests for AWS EKS Module. For now basic_flow commented out because of bug: https://github.com/epiphany-platform/m-aws-kubernetes-service/issues/32

### CODE REVIEW

File: integration_test.go [hunk 1]
```diff
@@ -0,0 +1,955 @@
...
+				t.Log("Resource group: ", rgName, " not found.")
+			} else {
+				t.Log("Resource group: Cannot get list of resources: ", errResourcesList)
+			}
+		} else {
+			t.Log("Resource group: There was an error: ", errResourcesList.Error())
+		}
+	}
+
+	resourcesTypesToRemove := []string{"Instance", "SecurityGroup", "NatGateway", "EIP", "InternetGateway", "Subnet", "RouteTable", "VPC"}
```
Comment (atsikham) [inline integration_test.go:568]: Probably the order should be changed, https://forums.aws.amazon.com/thread.jspa?threadID=92407

**Review by atsikham — CHANGES_REQUESTED** [review]

File: integration_test.go [hunk 2]
```diff
@@ -0,0 +1,955 @@
...
+		Region: aws.String(awsRegion),
+		Credentials: credentials.NewStaticCredentials(awsAccessKey, awsSecretKey, ""),
+	})
+	if errSession != nil {
+		t.Fatalf("Cannot get session: %s", errSession)
+	}
+
+	rgClient := resourcegroups.New(newSession)
+
+	rgName := moduleName + "-rg"
```
[the remark this answers is not in the document]
Comment (seriva) [reply integration_test.go:548]: Yes its correct as the code should remove resources used during testing which includes stuff from basic-infra module. Its similar that module and should be moved to a separate GO library down the road.

File: integration_test.go (hunk 1 above)
Comment (seriva) [reply integration_test.go:568]: Tried and tested code from basic-infrastructure module which I have no issues with so seems fine.

File: Makefile [hunk 3]
```diff
@@ -16,19 +29,40 @@ build: guard-VERSION guard-IMAGE guard-USER
 		-t $(USER)/$(IMAGE):$(VERSION) \
```
[the remark this answers is not in the document]
Comment (seriva) [reply Makefile]: Fixed

File: Makefile [hunk 4]
```diff
@@ -16,19 +29,40 @@ build: guard-VERSION guard-IMAGE guard-USER
...
+test: test-prerequisite \
+	build
+	@go test -v -timeout 90m
+
+test-release: test-prerequisite \
+	release
+	@CGO_ENABLED=0 go test -v -timeout 45m
+
+prepare-aws-credentials: guard-ACCESS_KEY guard-SECRET_KEY
+	@echo "$$AWS_CREDENTIALS_CONTENT" > $(ROOT_DIR)/awscreds.mk
```
[the remark this answers is not in the document]
Comment (seriva) [reply Makefile:42]: Nope as it contains credentials.

File: integration_test.go [hunk 5]
```diff
@@ -0,0 +1,955 @@
...
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/resourcegroups"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+)
+
+const (
+	awsbiImageTag = "epiphanyplatform/awsbi:0.0.1"
+	awsksImageTag = "epiphanyplatform/awsks:0.0.1"
```
[the remark this answers is not in the document]
Comment (seriva) [reply integration_test.go:32]: Yeah this needs probably some changes to facilitate running it on the pipeline.

File: Makefile [hunk 6]
```diff
@@ -16,19 +29,40 @@ build: guard-VERSION guard-IMAGE guard-USER
...
-	@docker run --rm \
-		-t $(USER)/$(IMAGE):$(VERSION) \
-		metadata
+test-prerequisite: guard-AWS_ACCESS_KEY guard-AWS_SECRET_KEY \
+		needs-docker \
+        needs-go \
+        needs-kubectl \
+		needs-aws
+
+needs-%:
```
[the remark this answers is not in the document]
Comment (mkyc) [reply Makefile:67]: smart

File: Makefile [hunk 7]
```diff
@@ -1,34 +1,68 @@
...
+	@echo "$$AWS_CREDENTIALS_CONTENT" > $(ROOT_DIR)/awscreds.mk
+
 release: guard-VERSION guard-IMAGE guard-USER
 	docker build \
 		--build-arg ARG_M_VERSION=$(VERSION) \
-		-t $(USER)/$(IMAGE):$(VERSION) \
+		-t $(IMAGE_NAME) \
 		.
 
+metadata: guard-VERSION guard-IMAGE guard-USER
```
Comment (mkyc) [inline Makefile:50]:
```suggestion
metadata: guard-VERSION guard-IMAGE guard-USER needs-docker
```

File: integration_test.go [hunk 8]
```diff
@@ -0,0 +1,1161 @@
...
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/cloudwatchlogs"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/eks"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/aws/aws-sdk-go/service/resourcegroups"
+	"github.com/go-test/deep"
+	"github.com/gruntwork-io/terratest/modules/docker"
+	//"github.com/gruntwork-io/terratest/modules/k8s"
```
Comment (mkyc) [inline integration_test.go:29]:
```suggestion
```

File: integration_test.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (mkyc) [reply integration_test.go:35]: yeah, this should be externalised. Possible way is to use environment variable and setup it [like this](https://github.com/epiphany-platform/cli/blob/develop/pkg/az/az_integration_test.go#L18-L35).

File: integration_test.go [hunk 9]
```diff
@@ -0,0 +1,1161 @@
...
+
+	cleanupPlan(t, "plan", sharedPath, awsAccessKey, awsSecretKey)
+	cleanupOutput(sharedPath)
+}
+
+//TODO: Enable when the following issue is fixed. 
+//      https://github.com/epiphany-platform/m-aws-kubernetes-service/issues/32
+/*
+func TestApply(t *testing.T) {
+	awsAccessKey, awsSecretKey := getAwsCreds(t)
```
Comment (mkyc) [inline integration_test.go:306]:
```suggestion
func TestApply(t *testing.T) {
	t.Skip() //TODO: Enable when the following issue is fixed: https://github.com/epiphany-platform/m-aws-kubernetes-service/issues/32
	awsAccessKey, awsSecretKey := getAwsCreds(t)
```

File: integration_test.go [hunk 10]
```diff
@@ -0,0 +1,1161 @@
...
+			}
+
+			docker.Run(t, awsksImageTag, destroyOpts)
+		})
+	}
+
+	cleanupPlan(t, "apply", sharedPath, awsAccessKey, awsSecretKey)
+	cleanupOutput(sharedPath)
+}
+*/
```
Comment (mkyc) [inline integration_test.go:406]:
```suggestion
```

File: integration_test.go [hunk 11]
```diff
@@ -0,0 +1,1161 @@
...
+	if len(awsSecretKey) == 0 {
+		t.Fatalf("expected non-empty AWS_SECRET_KEY environment variable")
+	}
+
+	return
+}
+
+//TODO: Move this to a separate GO library so we share it between the AWSBI and AWSEKS modules
+//      https://github.com/epiphany-platform/m-aws-kubernetes-service/issues/31
+const (
```
Comment (mkyc) [inline integration_test.go:535]: please move const section to head of file

**Review by mkyc — CHANGES_REQUESTED** [review]
Rather little comments

File: integration_test.go (hunk 9 above)
Comment (seriva) [reply integration_test.go]: Good one:)

File: Makefile (hunk 6 above)
[the remark this answers is not in the document]
Comment (seriva) [reply Makefile:67]: thank @sk4zuzu

File: integration_test.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (seriva) [reply integration_test.go]: Made them available externally through env variables.

**Review by mkyc — APPROVED** [review]
LGTM

**Review by atsikham — APPROVED** [review]
LGTM

**Pull request merged**